### PR TITLE
feat(map/basic): added hover function

### PR DESCRIPTION
**Pull request by juanmendez90**
Fix hover functionality, now if you step in a poi the polygon maintains its background-color.
The problem I am trying to solve here is that when you hover a POI on the map the hover class of the polygon is lost

### CODE REVIEW

File: components/map/basic/src/leaflet/map.js [hunk 1]
```diff
@@ -162,7 +163,123 @@ export default class LeafletMap {
     }
   }
 
+  pnpoly(x, y, coords) {
```
Comment (midudev) [inline components/map/basic/src/leaflet/map.js]: Is this magic? What's this doing?

File: components/map/basic/src/leaflet/map.js [hunk 2]
```diff
@@ -162,7 +163,123 @@ export default class LeafletMap {
...
+    this._map.on('mousemove', ({latlng}) => {
+      const geoJSON = this.layerManager.getLayers(
+        this.layerManager.layers.geoJSON
+      )
+      geoJSON.forEach((item, index) => {
+        const point = {coordinates: [latlng.lng, latlng.lat], type: 'Point'}
+        item.eachLayer(layer => {
+          const {_path: path = {}} = layer
+          const {classList} = path
+          classList && classList.remove('re-Mappage-polygon-hover')
```
Comment (midudev) [inline components/map/basic/src/leaflet/map.js]: This is strongly attached to Fotocasa needs.

**Comment by midudev** [comment]
Can we have more context about the problem that this is trying to solve? Tbh, I'm hardly able to follow the code :D

File: components/map/basic/src/leaflet/map.js (hunk 1 above)
Comment (juanmendez90) [reply components/map/basic/src/leaflet/map.js]: I don't know 🤣, this was copy-paste from here https://github.com/maxogden/geojson-js-utils/blob/aa31043f31cb65d573a29dfc9faef49f12702b7c/geojson-utils.js#L72

File: components/map/basic/src/leaflet/map.js (hunk 2 above)
Comment (juanmendez90) [reply components/map/basic/src/leaflet/map.js]: the map it self is strongly attached to fotocasa need. Should I send these classes via props?

**Comment by midudev** [comment]
Closing as other PR was used for fixing this.

**Pull request closed without merging**